lib/compat/getent.h

```diff
@@ -28,21 +28,21 @@
 
 #ifndef SYSLOG_NG_HAVE_GETPROTOBYNUMBER_R
 
+#include "getent-generic.h"
+
 #define getprotobynumber_r _compat_generic__getprotobynumber_r
 #define getprotobyname_r _compat_generic__getprotobyname_r
 #define getservbyport_r _compat_generic__getservbyport_r
 #define getservbyname_r _compat_generic__getservbyname_r
 
-#include "getent-generic.h"
-
 #elif defined(sun) || defined(__sun)
 
+#include "getent-sun.h"
```

Comment:
Can you explain the root cause and why the reordering fixes the problem?
Please update the commit message accordingly.

Reply:
Thanks for the tip, done :)